src/main/java/com/puppycrawl/tools/checkstyle/checks/AvoidEscapedUnicodeCharactersCheck.java

```diff
@@ -348,7 +350,8 @@ public void beginTree(DetailAST rootAST) {
 
     @Override
     public void visitToken(DetailAST ast) {
-        final String literal = ast.getText();
+        final String literal =
+            CheckUtil.stripIndentAndInitialNewLineFromTextBlock(ast.getText());
```

Comment:
Confirm Issue https://github.com/checkstyle/checkstyle/issues/9159 represents this fix. I will approve that issue and you should split this into 2 commits, 1 for new functionality and 1 for a bug fix. Then we can merge it since others have all approved this.

Reply:
Issue #9159 does represent this fix. PR split.